qemu/tests/block_hotplug_in_pause.py

```diff
@@ -0,0 +1,201 @@
+        """
+        image_params = params.object_params(image_name)
+        devs = vm.devices.images_define_by_params(image_name,
+                                                  image_params, 'disk')
+        for dev in devs:
+            ret = vm.devices.simple_hotplug(dev, vm.monitor)
+            if ret[1] is False:
+                test.fail("Failed to hotplug device '%s'."
+                          "Output:\n%s" % (dev, ret[0]))
+        devs = [dev for dev in devs if not isinstance(dev, qdevices.QDrive)]
+        return devs
+
+    def block_unplug(device_list):
+        """
+        Unplug disks and verify it in qtree
+
+        :param device_list: List of objectes for unplug disks
+        """
+        for dev in reversed(device_list):
+            ret = vm.devices.simple_unplug(dev, vm.monitor)
+            if ret[1] is False:
+                test.fail("Failed to unplug device '%s'."
+                          "Ouptut:\n%s" % (dev, ret[0]))
+
+    def block_check_in_guest(session, disks, blk_num,
+                             get_disk_cmd, plug_tag="hotplug"):
+        """
+        Check hotplug/unplug disks in guest
+
+        :param session: A shell session object.
+        :param disks: List of disks before hotplug/unplug.
+        :param blk_num: Number of hotplug/unplug disks.
+        :param get_disk_cmd: Cmd to get disks info in guest.
+        :plug_tag: Tag for hotplug/unplug
+        """
+        logging.info("Check block device in guest after %s." % plug_tag)
+        pause = float(params.get("virtio_block_pause", 30.0))
+        status = utils_misc.wait_for(lambda: len(get_plug_unplug_disks(disks,
+                                     find_disk(session, get_disk_cmd))) == blk_num,
+                                     pause)
+        disks = get_plug_unplug_disks(disks, find_disk(session, get_disk_cmd))
+        if not status:
+            test.fail("Failed to %s device to guest, expected: %d,"
+                      "actual: %d" % (plug_tag, blk_num, len(disks)))
+
+    def rw_disk_in_guest(session, plug_disks, iteration):
+        """
+        Do read/write on hotplug disks
+
+        :param session: A shell session object.
+        :param plug_disks: List for hotplug disks
+        :iteration: repeat times for hotplug.
+        """
+        logging.info("Read/Writ on block device after hotplug.")
+        disk_op_timeout = int(params.get("disk_op_timeout", 360))
+        disk_letter = params.objects("disk_letter")
+        for disk in plug_disks:
+            if params.get("os_type") == "linux":
+                test_cmd = params.get("disk_op_cmd") % (disk, disk)
+            elif params.get("os_type") == "windows":
+                if iteration == 0:
+                    error_context.context("Format disk", logging.info)
+                    did = disk.split()[0]
+                    size = disk.split()[1] + "B"
+                    disk = utils_disk.configure_empty_windows_disk(session, did, size)[0]
```

Comment:
need clear readonly status of new added disk first, before run configure_empty_windows_disk(), like follows commands:
 if not utils_disk.update_windows_disk_attributes(session, disk):
      test.fail("Failed to update windows disk attributes.")
otherwise, will report error:
22:39:43 ERROR| ERROR 1-Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_hotplug_in_pause.with_plug.one_pci.q35 -> ShellStatusError: Could not get exit status of command: 'echo list disk > disk_r3Hx && echo select disk 1 >> disk_r3Hx &&  echo create partition primary size=1014 >> disk_r3Hx &&  echo exit >> disk_r3Hx && diskpart /s disk_r3Hx && del /f disk_r3Hx'    (output: '\nMicrosoft DiskPart version 10.0.14393.0\n\nCopyright (C) 1999-2013 Microsoft Corporation.\nOn computer: WIN-9SHHPMLFOOC\n\n  Disk ###  Status         Size     Free     Dyn  Gpt\n  --------  -------------  -------  -------  ---  ---\n  Disk 0    Online           30 GB      0 B         \n  Disk 1    Offline        1024 MB  1024 MB         \n\nDisk 1 is now the selected disk.\n\nDiskPart has encountered an error: The media is write protected.\nSee the System Event Log for more information.\n\n')